atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/keyvalue/KeyValueServiceCoordinationStore.java

```diff
@@ -0,0 +1,201 @@
+import com.palantir.atlasdb.table.description.ColumnMetadataDescription;
+import com.palantir.atlasdb.table.description.ColumnValueDescription;
+import com.palantir.atlasdb.table.description.DynamicColumnDescription;
+import com.palantir.atlasdb.table.description.NameComponentDescription;
+import com.palantir.atlasdb.table.description.NameMetadataDescription;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.remoting3.ext.jackson.ObjectMappers;
+
+import okio.ByteString;
+
+/**
+ * An implementation of {@link CoordinationStore} that persists its state in a {@link KeyValueService}.
+ *
+ * Specifically, a {@link KeyValueServiceCoordinationStore} stores its data in a single row in the
+ * {@link AtlasDbConstants#COORDINATION_TABLE}. This specific row is identified by the store's
+ * {@link KeyValueServiceCoordinationStore#coordinationSequence}; the reason for doing this is that it allows
+ * for multiple distinct sequences to coexist in one table.
+ *
+ * Within a single identified row, values are written with dynamic column keys as sequence numbers, which are
+ * positive longs. Furthermore, a {@link SequenceAndBound} which controls which of the values is considered to be
+ * the most recent in the coordination sequence and how long it is valid for is written at dynamic column key zero.
+ * This {@link SequenceAndBound} should contain the ID of the value that is considered to be the most current
+ * in the sequence.
+ */
+// TODO (jkong): Coordination stores should be able to clean up old values.
+public final class KeyValueServiceCoordinationStore implements CoordinationStore {
+    private static final Logger log = LoggerFactory.getLogger(KeyValueServiceCoordinationStore.class);
+
+    private static final TableMetadata COORDINATION_TABLE_METADATA = new TableMetadata(
+            NameMetadataDescription.create(ImmutableList.of(
+                    new NameComponentDescription.Builder()
+                            .componentName("sequence")
+                            .type(ValueType.BLOB)
+                            .logSafety(TableMetadataPersistence.LogSafety.SAFE)
+                            .build())),
+            new ColumnMetadataDescription(new DynamicColumnDescription(
+                    NameMetadataDescription.create(
+                            ImmutableList.of(
+                                    new NameComponentDescription.Builder()
+                                            .componentName("sequenceNumber")
+                                            .logSafety(TableMetadataPersistence.LogSafety.SAFE)
+                                            .type(ValueType.VAR_LONG)
+                                            .build())),
+                    ColumnValueDescription.forType(ValueType.VAR_LONG))),
```

Comment:
Should this be `BLOB`? Sequence values certainly seem to be blobs, and the special sequence number 0 is a `SequenceAndBound`.